src/subdataframe/subdataframe.jl

```diff
@@ -51,7 +51,7 @@ immutable SubDataFrame{T <: AbstractVector{Int}} <: AbstractDataFrame
     parent::DataFrame
     rows::T # maps from subdf row indexes to parent row indexes
 
-    function SubDataFrame(parent::DataFrame, rows::T)
+    @compat function SubDataFrame{T}(parent::DataFrame, rows::T) where T
```

Comment:
Not sure. I just followed the deprecation warning. Shouldn't harm to make that restriction though.